src/lib/chainNode.ts

```diff
@@ -171,60 +171,47 @@ export default class ChainNode {
     return signed
   }
 
-  async submitRunProcess(
+  submitRunProcess(
     extrinsic: SubmittableExtrinsic<'promise', SubmittableResult>,
     transactionDbUpdate: (state: TransactionState) => Promise<void>
-  ): Promise<number[]> {
+  ): void {
     this.logger.debug('Submitting Transaction %j', extrinsic.hash.toHex())
-    return new Promise((resolve, reject) => {
-      let unsub: () => void
```

Comment:
we still need `unsub` as it was before. Otherwise the callback we pass will never be removed from an internal event handler in `polkadot.js`

Reply:
this has been added as per our conversation.